### PR TITLE
Async mode fix

### PR DESCRIPTION
**Pull request by sepandhaghighi**
#### Reference Issues/PRs
#5 #27 #26 
#### What does this implement/fix? Explain your changes.

- `NavaThread` class added
- `stop` function added
- `stop_all` function added
- `autopep8` scripts updated
- `README.md` updated
- `CHANGELOG.md` updated

### CODE REVIEW

**Comment by sadrasabouri** [comment]
## Local tests on OSs
- [x] MacOS Sonoma 14.2.1

File: nava/thread.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    def stop(self):
+        """
+        Stop sound.
+
+        :return: None
+        """
+        sys_platform = sys.platform
+        if sys_platform == "win32":
+            import winsound
+            winsound.PlaySound(None, winsound.SND_PURGE)
```
Comment (sadrasabouri) [inline nava/thread.py:41]: Did you test the case in Windows where you have multiple sounds playing asynchronously and try to stop just one of them?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,11 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- Async mode
```
Comment (sadrasabouri) [inline CHANGELOG.md:9]: I think "`async_mode` parameter" under `Changed` is better.

File: nava/functions.py [hunk 3]
```diff
@@ -5,17 +5,46 @@
...
+
+
+def sound_id_gen():
+    """
+    Sound id generator.
+
+    :return: sound id as int
+    """
+    params._play_threads_counter += 1
+    sound_id = params._play_threads_counter + 1000
```
Comment (sadrasabouri) [inline nava/functions.py:46]: Why did you add `1000` offset. If it's necessary, what do you think of transferring it to `params`.

File: nava/functions.py [hunk 4]
```diff
@@ -53,137 +82,123 @@ def quoter(sound_path, *args, **kwargs):
...
+    :return: None or sound id
     """
-    for proc in play_processes:
-        proc.kill()
-        proc.terminate()
+    import winsound
+    play_flags = winsound.SND_FILENAME | (async_mode & winsound.SND_ASYNC)
+
+    if async_mode:
+        sound_thread = NavaThread(target=__play_win_flags, args=(sound_path, play_flags), daemon=True)
```
Comment (sadrasabouri) [inline nava/functions.py:99]: It's too long. Can you make it multiline?

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
Thanks for your efforts. That was a great job.

File: nava/thread.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    def __init__(self, *args, **kwargs):
+        """
+        Init method.
+
+        :param args: arguments
+        :type args: list
+        :param kwargs: keyword arguments
+        :type kwargs: dict
+        """
+        super(NavaThread, self).__init__(*args, **kwargs)
```
[the remark this answers is not in the document]
Comment (sepandhaghighi) [reply nava/thread.py:20]: It will help `NavaThread` to act like the original `Thread` in other aspects.

File: nava/thread.py (hunk 1 above)
Comment (sepandhaghighi) [reply nava/thread.py:41]: Currently, the `winsound` module cannot play multiple sounds simultaneously. But play/stop of one sound in async mode works like charm!

File: nava/functions.py (hunk 3 above)
Comment (sepandhaghighi) [reply nava/functions.py:46]: For better readability, we may change the format of the `sound_id` to something like `Nava-1`, `Nava-2`, etc. in the future.

File: nava/functions.py (hunk 4 above)
Comment (sepandhaghighi) [reply nava/functions.py]: fixed --> https://github.com/openscilab/nava/pull/29/commits/b0f52fcbda301d406a0ba456c779ac3575459f07

File: nava/thread.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (sadrasabouri) [reply nava/thread.py:20]: Oh, I see. It seems a high-level trick. 🥇

**Review by sadrasabouri — APPROVED** [review]
Well done. Thanks
